enhancements/insights/conditional-data-gathering.md

```diff
@@ -0,0 +1,231 @@
+
+### User Stories
+
+For OpenShift engineers it will be easier to get necessary information when some problem occurs (like alert is firing).
+Getting such information now means changing operator's code which is especially painful with older versions of clusters.
+When the data is not needed anymore, the rule can be simply removed from the repo.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Empty
+
+### Risks and Mitigations
+
+The potential risks could come from an attacker spoofing the config somehow (if they got access to the repo),
+all they could do is let the insights operator collect more data and send it to the c.r.c, but because of validation
+on the operator side, the potential of collecting data is limited and there still would be
+the same anonymization of potentially sensitive data as before.
+For example, we check that namespaces start with `openshift-`, we're also limiting
+the amount of potentially collected data by, for example, introducing the limit for amount of collected logs
+(per container and the number of containers) and, in the worst case, if the conditional gathering takes too much time
+it would be stopped by the timeout and we would just get less data.
+
+Also in the regular workflow, changing the config involves going through the repo's CI (JSON schema validator)
+and probably a simple review process.
+
+Not really a risk, but in case the service is not available or provides invalid config for the operator,
+we would just have less data in the archives (everything except conditional gatherer's data),
+an error would be written to the logs and to the metadata which would allow us to know
+that something is broken and we need to bring the service back.
+
+In case GitHub is not available, we won't be able to add new rules, but the old ones would still be returned by
+the service.
+
+## Design Details
+
+Empty
+
+### Open Questions [optional]
+
+Empty
+
+### Test Plan
+
+The conditional gatherer is covered with unit tests considering many different scenarios.
+Later there will be integration tests as well.
+
+### Graduation Criteria
+
+Empty
+
+#### Dev Preview -> Tech Preview
+
+Empty
+
+#### Tech Preview -> GA
+
+Empty
+
+#### Removing a deprecated feature
+
+Empty
+
+### Upgrade / Downgrade Strategy
+
+The conditional gatherer itself (e.g. adding new/removing old conditions or gathering functions) is
+modified through a standard process like all other operator's code. Only the rules to collect X with Y params on Z
+are updated through the new process described above.
+
+### Version Skew Strategy
+
+Out of scope.
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Drawbacks
+
+None?
+
+## Alternatives
+
+Thin IO idea (https://github.com/openshift/enhancements/pull/465) could also solve this, but thin IO was about adding
+new code bypassing the standard release process which could potentially be very dangerous.
```

Comment:
This proposal is keeping the rule development in the existing insights-operator repository.  But then it's delivering them via a new, central console.redhat.com API, so you can end up with new-rule-vs-older-insights-operator-code combinations that have not been tested by CI/QE.  And I don't see anything addressing [the signing and restricted-network concerns][1] raised in the previous PR.  The OLM approach floated in previous discussion seems like it would cover both of those without needing a new API.

[1]: https://github.com/openshift/enhancements/pull/465#issuecomment-770977399

Reply:
Well, there will be basic tests, but yes, we can't test every single combination. The Thin IO was very different, it was about running new code, here we have only freedom to run existing functions with different validated parameters.